src/zenml/client.py

```diff
@@ -5506,7 +5585,7 @@ def get_model_version(
         model_version_name_or_number_or_id: Optional[
             Union[str, int, UUID, ModelStages]
         ] = None,
-    ) -> ModelVersionResponseModel:
+    ) -> "ModelVersion":
```

Comment:
It is a getter method, so it returns, delete returns nothing and it is a few lines above.

Regarding your major concern here: this whole PR is around making the client and context return the same object the user used for the configuration of a pipeline. So we improve the learning curve, simplify the experience, and make a distinction between user-facing friendly objects and crazy Response models, which one can hardly navigate due to complexity (at least before hydration work was a thing at all).

Reply:
I believe the main goal of the PR was to replace the `ModelConfig` object when configuring pipelines / ... This does not necessarily mean that the `ModelVersion` should be used wherever possible. In the case of the Client, AFAIK all other CRUD methods return response models, so I think it would make more sense to keep returning `ModelVersionResponseModel` there and to do the conversion to `ModelVersion` outside of the Client wherever it's necessary

Reply:
Prefer to keep it as is, no one even knows about `zenml.models.model_models.ModelVersionResponseModel` thing - it is too internal to be a concerning, in my opinion.